enhancements/worker-latency-profile/worker-latency-profile.md

```diff
@@ -0,0 +1,421 @@
+
+### Goals
+
+* Enable Openshift cluster to fine tune the reliability in medium to high latency scenarios by specifying simple `WorkerLatencyProfile` at day-0 as well as day-2.
+
+### Non-Goals
+
+* Modify the existing `Kubelet`, `Kube Controller Manager` or `Kube API Server` code in any way.
+* Allow any kind of different latency scenarios between masters
+
+### User Stories
+
+* User wants to fine tune the cluster reliability for their node latency scenario while making sure they are protected from setting parameters that could potentially break the cluster.
+
+## Proposal
+
+* The option to set `WorkerLatencyProfile` will have to reside in a centralized location. [Recently proposed Node object](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/mco-cgroupsv2-support.md#api-extensions) in `config/v1/types_node.go` would the ideal place to define API extensions necessary for `WorkerLatencyProfile`.
+* [Machine Config Operator (MCO)](https://github.com/openshift/machine-config-operator) sets the appropriate value of the `Kubelet` flag `--node-status-update-frequency`
+* [Kubernetes Controller Manager operator (KCMO)](https://github.com/openshift/cluster-kube-controller-manager-operator) sets the appropriate value of the `Kube Controller Manager` flag `--node-monitor-grace-period`
+* [Kubernetes API Server Operator (KASO)](https://github.com/openshift/cluster-kube-apiserver-operator) sets the appropriate value of the `Kube API Server` flags `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds`
+
+## Design Details
+
+### API Extensions
+
+The [existing proposed node object](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/mco-cgroupsv2-support.md#api-extensions) needs to be updated to include `NodeStatus` to track the progress of `WorkerLatencyProfile` rollout.
+
+```go
+type Node struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec NodeSpec `json:"spec"`
+
+  // status holds observed values.
+	// +optional
+	Status NodeStatus `json:"status"`
+}
+
+type NodeSpec struct {
+  WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+  // an eventual additional option might be crun in the future. This explains
+  //   why a new struct may be necessary
+  // CgroupMode CgroupMode `json:"cgroupMode,omitempty"`
+  //
+  // CrunEnabled bool ...
+}
+
+type NodeStatus struct {
+  // WorkerLatencyProfileStatus provides the current status of WorkerLatencyProfile
+  WorkerLatencyProfileStatus WorkerLatencyProfileStatus `json:"workerLatencyProfileStatus,omitempty"`
+}
+
+type WorkerLatencyProfileType string
+
+const (
+    // Medium Update and Average Reaction
+    MediumUpdateAndAverageReaction WorkerLatencyProfileType = "MediumUpdateAndAverageReaction"
```

Comment:
Nit: drop the and from the name and make it just `MediumUpdateAverageReaction` and `LowUpdateSlowReaction` and provide a better description above the field, ie: `Medium Kubelet Update Frequency (heart-beat) and Average Reaction Time to unresponsive Node` same the other one.